packages/brick_offline_first_with_rest/lib/src/offline_queue/rest_offline_queue_client.dart

```diff
@@ -19,6 +19,14 @@ class RestOfflineQueueClient extends http.BaseClient {
 
   final RequestSqliteCacheManager<http.Request> requestManager;
 
+  /// A callback triggered when the response of a request has a status code
+  /// which is present in the [reattemptForStatusCodes] list.
+  void Function(http.Request request, http.StreamedResponse response)?
+      onReattemptableResponse;
```

Comment:
Would prefer to keep the names similar
```suggestion
  /// A callback triggered when the response of a request has a status code
  /// which is present in the [reattemptForStatusCodes] list.
  void Function(http.Request request, http.StreamedResponse response)?
      onReattemptResponse;
```

Reply:
Is the `onReattemptResponse` necessary? Your PR description notes that it can help resolve a blocked queue, but is it clear how many retries have been made on this request? Is that necessary to determine when to skip a request? If you've explicitly told Brick "reattempt in this scenario" and now you want to no longer attempt for that scenario, shouldn't you remove the status code from the reattempt list instead?

Reply:
The `onReattemptResponse` helps to
1. know that the client is online (because there was a response from the server)
2. check if the status code of the returned response code indicates that the server rejected the response (e.g., HTTP 409) (in which case I prompt the user to handle the conflict)

>  If you've explicitly told Brick "reattempt in this scenario" and now you want to no longer attempt for that scenario, shouldn't you remove the status code from the reattempt list instead?

If I removed the status code, (e.g., HTTP 409) from the reattempt from status codes list, the client had no chance to detect and handle the rejected response and the data is lost.

Reply:
> Would prefer to keep the names similar

I think naming the callback `onReattemptResponse` is misleading, as it sounds like a response would be reattempted. But my initial suggestion `onReattemptableResponse` has the same problem.

What about `onReattemptableRequestFailed`?

Reply:
What about `onRequestFailure`?

Reply:
So you would go with `onRequestFailure` and `onRequestError`? Then it's difficult to guess from the name which one receives which callback.

What about 
`onReattemptableRequestFailure` and `onRequestException` or 
`onResponse` and `onRequestException`?